arduino/cores/board.go

```diff
@@ -139,3 +139,43 @@ func (b *Board) GeneratePropertiesForConfiguration(config string) (*properties.M
 	}
 	return b.GetBuildProperties(fqbn.Configs)
 }
+
+// IsBoardMatchingIDProperties returns true if the board match the given
+// identification properties
+func (b *Board) IsBoardMatchingIDProperties(query *properties.Map) bool {
+	portIDPropsSet := b.Properties.SubTree("upload_port")
+	if portIDPropsSet.Size() == 0 {
+		return false
+	}
+
+	check := func(p *properties.Map) bool {
+		for k, v := range p.AsMap() {
+			if !strings.EqualFold(query.Get(k), v) {
+				return false
+			}
+		}
+		return true
+	}
+
+	idx := 0
+	haveIndexedProperties := false
+	for {
+		idProps := portIDPropsSet.SubTree(fmt.Sprintf("%d", idx))
+		idx++
+		if idProps.Size() > 0 {
+			haveIndexedProperties = true
+			if check(idProps) {
+				return true
+			}
+		} else if idx > 0 {
+			// Always check sub-id 0 and 1 (https://github.com/arduino/arduino-cli/issues/456)
+			break
+		}
+	}
+
+	if !haveIndexedProperties {
+		return check(portIDPropsSet)
+	}
```

Comment:
Couldn't this be done first? I'd expect platforms that support the new `upload_port` syntax to not have indexed properties too.

Reply:
Nevermind, I saw below that you conver `vid.0` and the likes to `upload_port.0` so we still support it.